src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -2851,16 +2852,16 @@ func (p *writeBatchPooledReqPool) Init() {
 
 func (p *writeBatchPooledReqPool) Get(size int) *writeBatchPooledReq {
 	cappedSize := size
-	if cappedSize > client.DefaultWriteBatchSize {
-		cappedSize = client.DefaultWriteBatchSize
+	if cappedSize > (2 * client.DefaultWriteBatchSize) {
```

Comment:
can we please add a comment on why this is required? 
and also please refer the commit where it was already similar to this but got changed?

Reply:
Sure have added a comment. Also leaving here the previous commit which had the same size set. The only difference now is that this time it is adaptive to the incoming batch size as well.

https://github.com/m3db/m3/pull/677/files#diff-8bea1e27411e7182c9db15cae2ee3cbb3d4e55d8ba57215cdf2e9231b7491abdR1391

Reply:
I'm not really sure about the need for this 2x increase. When searching for call sites of newPooledID, I only see this id pooling being used for namespace and series ids, but not for encoded tags:
<img width="939" alt="Screenshot 2023-09-09 at 20 46 12" src="https://github.com/m3db/m3/assets/5231884/d586e82b-6c82-4c96-933f-78f3e77fb5bd">

Reply:
BTW you can check the average batch size in your system with PromQL query such as
```
(sum(rate(service_writeTaggedBatchRaw_success[1m])) + sum(rate(service_writeTaggedBatchRaw_errors[1m]))) / sum(rate(service_writeTaggedBatchRaw_rpcs[1m]))
```

Reply:
Thank you @linasm for the correction, I still think if we can stick with 2x increase, it would be helpful for the clients which doing heavy writes and are using write batch size greater than the default. In our case the average batch size is above 450, with our default write batch size in the client is 512. 
When we move back from the 2x to x, we see significant spikes in P99 latencies.
The code will work the same for the clients who are using the default write batch size of 128.

I have updated the comment as well.
Does this look fine?

Reply:
![image](https://github.com/m3db/m3/assets/19673740/6bcde305-29be-4b41-a248-2ce4d176ea84)

This shows the latency difference between the two versions, @linasm , hope this helps as well